wallet/chainntfns.go

```diff
@@ -38,6 +38,22 @@ func (w *Wallet) handleConsensusRPCNotifications(chainClient *chain.RPCClient) {
 			err = walletdb.Update(w.db, func(tx walletdb.ReadWriteTx) error {
 				return w.onBlockConnected(tx, n.BlockHeader, n.Transactions)
 			})
+
+			if w.StakeMiningEnabled && w.purchaser != nil {
```

Comment:
I think this also shows that the current design is backwards.  wallet should not call the ticket purchaser.  it should emit an event that causes the started ticket purchaser goroutine (if any) to begin purchasing tickets.